jump/src/installer.rs

```diff
@@ -135,10 +152,46 @@ pub(crate) fn install(files: &[FileEntry], payload: &[u8]) -> Result<(), String>
                     scie_tote.push((file, file.file_type, dst.clone()));
                 } else {
                     let bytes = &payload[location..(location + file.size)];
-                    unpack(file.file_type, Cursor::new(bytes), file.hash.as_str(), dst)?;
+                    unpack(
+                        file.file_type,
+                        || Ok((Cursor::new(bytes), ())),
+                        file.hash.as_str(),
+                        dst,
+                    )?;
                 }
                 file.size
             }
+            FileEntry::LoadAndInstall((binding, file, dst)) => {
+                let buffer_source = || {
```

Comment:
This is the core feature, if a file in the lift manifest has a "source" field, it's expected to be the name of a binding and that binding command is expected to take the name of the file to load as its sole argument and then emit the contents of the file on stdout. The extra lift manifest metadata is the bridge from file names to urls (or whatever extra metadata the load binding needs to know to load the file).

Reply:
@benjyw and @sureshjoshi - this is the key interface I'm hoping for feedback on.

Reply:
@benjyw and @sureshjoshi I can proceed without you, but I was hoping for a sanity check on at least this key interface. The driver here is not bloating the scie-jump but still allowing opt-in flexibility in scie construction by those willing to take the hit of embedding a utility to materialize files with. I'll likely proceed getting this PR in by evening.

Reply:
Was indeed just looking at this, sorry for the delay! See comments on interface.

Reply:
I spent the last couple days of OSS time landing some other PRs in Pants and others. I can look at this, this evening briefly - but I think Benjy's covering a lot of my thoughts as well already

Reply:
Yeah, I do appreciate both of your time looking at this. I debate man in bus vs stealing attention.  There is nothing more valuable than attention and stealing it should be considered seriously.

So the motivating use case mentioned in #19 was a Twitter use case where shipping around very very large binaries was a hindrance on the dev cycle. It was much faster to ship an app descriptor from a laptop and then have the elements assembled "on-site" in the fast network environment of the production datacenter. This serves that case but also generally allows you to avoid waste. For example, if Pants ships as a scie, every new version of Pants will include a copy of the same ~64MB embedded Python interpreter. That interpreter will only extract once from the 1st Pants scie using that interpreter - say Python 3.10. The rest of the Pants binaries will just lug it along for no real reason. Certainly a small sin by today's standards, but one nonetheless. You could imagine instead shipping a thin scie instead without the Python interpreter embedded but instead fetched if needed (not already in the local machine `~/.nce` cache).

Now that's roughly the motivating set of use cases here. The mechanism is a bit more general though - I'm not sure what people might do with it in terms of loading files. You could imagine pulling them them from other sources like a secured vault on the local machine allowing you to avoid distributing sensitive content.

Reply:
Okay, thanks great context for this.

I can think up some use cases where I could use it, but practically speaking - I wouldn't just because of all of the edge cases that would inevitably kick me in the ass, because I'm the epitome of Murphy's Law. 

Very useful from an app-update point of view though - especially in the embedded world, where disk space is genuinely at a premium.